src/RocketChatAttachment.php

```diff
@@ -103,18 +102,25 @@ public function text(string $text): self
     }
 
     /**
-     * @param string|\DateTime $timestamp
+     * @param string|\DateTimeInterface $timestamp
      * @return \NotificationChannels\RocketChat\RocketChatAttachment
      */
     public function timestamp($timestamp): self
     {
-        if (! ($timestamp instanceof DateTime) && ! is_string($timestamp)) {
-            throw new InvalidArgumentException('Timestamp must be string or DateTime, '.gettype($timestamp).' given.');
+        if (! ($timestamp instanceof DateTimeInterface) && ! is_string($timestamp)) {
+            $invalidType = is_object($timestamp)
+                ? get_class($timestamp)
+                : gettype($timestamp);
+
+            throw new InvalidArgumentException(sprintf(
+                'Timestamp must be string or DateTime, %s given.',
+                $invalidType
+            ));
         }
 
-        if ($timestamp instanceof DateTime) {
+        if ($timestamp instanceof DateTimeInterface) {
             $date = clone $timestamp;
```

Comment:
I think we don't need clone here, because we are not mutating object anymore. We could just format it.